From c459b52c0f300860bb877b46165018a882e8ee99 Mon Sep 17 00:00:00 2001
From: Ben Adams <thundercat@illyriad.co.uk>
Date: Thu, 11 Jan 2018 09:10:35 +0000
Subject: [PATCH] Allow CALLEE_IS_FORCE_INLINE precedent over
 CALLEE_DOES_NOT_RETURN (#14586)

Allow CALLEE_IS_FORCE_INLINE precedent over CALLEE_DOES_NOT_RETURN
---
 src/jit/inlinepolicy.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/jit/inlinepolicy.cpp b/src/jit/inlinepolicy.cpp
index 959ef1b..306a04d 100644
--- a/src/jit/inlinepolicy.cpp
+++ b/src/jit/inlinepolicy.cpp
@@ -484,7 +484,8 @@ void DefaultPolicy::NoteInt(InlineObservation obs, int value)
 
             unsigned basicBlockCount = static_cast<unsigned>(value);
 
-            if (m_IsNoReturn && (basicBlockCount == 1))
+            // CALLEE_IS_FORCE_INLINE overrides CALLEE_DOES_NOT_RETURN
+            if (!m_IsForceInline && m_IsNoReturn && (basicBlockCount == 1))
             {
                 SetNever(InlineObservation::CALLEE_DOES_NOT_RETURN);
             }
-- 
2.7.4

